### ISSUE TITLE
Custom validators are not getting run on array fields

### ISSUE THREAD

**Issue by rmacqueen** [Automattic/mongoose#4039]
During `findOneAndUpdate()`, custom validators are not getting run on schema fields where the field type is an array.
For example, this code snippet: 

```
var mongoose = require('mongoose');
mongoose.connect('mongodb://localhost/myTestDB');
 
var db = mongoose.connection;
 
db.on('error', function (err) {
console.log('connection error', err);
});
db.once('open', function () {
console.log('connected.');
});
 
var Schema = mongoose.Schema;
var userSchema = new Schema({
  _id : String,
  name : {
    type: [String],
    validate: {
        validator: function (str) {
            return str.length > 1
        }
    },
  }
});


var User = mongoose.model('User', userSchema);

User.findOneAndUpdate({"name": ["John", "Doe"]}, { 
  $setOnInsert: {
    name: ["John"],
  },
}, { runValidators: true, upsert: true, new: true }, function (err, data) {
  if (err) {
    return console.log(err);
  } else {
    // console.log(data.validateSync())
    return console.log('Updated', data);
  }
});
```

will allow you to update the user to have a `name` field of ["John"] without throwing any errors, even though the custom validator I included explicitly prohibits any name array of length less than or equal to 1. The validator itself works just fine, as can be seen by the fact that if you uncomment the line `console.log(data.validateSync())`, thereby forcing the validation to take place, it will in fact return the appropriate error message. The problem is that this validation is not taking place within the `findOneAndUpdate()` call, despite the fact that I include the `runValidators=true` option.

node version: 3.1.0
mongodb version: 3.0.6
mongoose version: 4.1.5

**Comment by vkarpov15** [comment]
@rmacqueen upgrade to [mongoose >= 4.3.5](https://github.com/Automattic/mongoose/blob/master/History.md#435--2016-01-09), this was fixed with #3724.

**Comment by vkarpov15** [comment]
@Saravanan90 please open up a new issue with code samples
